src/components/organization/PartnerForm.js

```diff
@@ -0,0 +1,108 @@
+import {Button, Grid, makeStyles, TextField} from "@material-ui/core";
+import {Controller, useForm} from "react-hook-form";
+import {validateEmail} from "../../utils/functions";
+import React, {useState} from "react";
+import {withSwal} from "../formsCommons/WithSwal";
+import {organizationsAPI} from "../../utils/apis/OrganizationApi";
+
+const useStyles = makeStyles((theme) => ({
+    button: {
+        height: '97%'
+    }
+}))
+
+export const PartnerForm = (props) => {
+
+    const classes = useStyles()
+
+    const defaultFormState = {
+        firstName: "",
+        lastName: "",
+        mail: ""
+    }
+
+    const [users, setUsers] = useState([props.users])
+
+    const {handleSubmit, control, reset} = useForm({defaultValues:defaultFormState})
+
+    const onUserAdd = (data) => {
+        withSwal({
+            loadingTitle: "Adding HR user",
+            promise: () => organizationsAPI.addHrPartner(data),
+            successSwalTitle: "Success",
+            successSwalText: "You've successfully added HR user!",
+            successFunction: () => {
+                setUsers(users.concat(data))
+                reset()
+                if(props.onSubmit){
+                    props.onSubmit(data)
+                }
+                props.userAdded(data)
+            },
+            errorSwalTitle: "We couldn't save this user for you"
+        })
+
+    }
+
+    return <>
+        <form id="user-form" name="user-form" onSubmit={handleSubmit(onUserAdd)}/>
```

Comment:
Na pewno tak to wywoływaliśmy gdzieś indziej? Ogólnie as far as I remember, tak wywołujesz tę funkcję, więc raczej 2/10, ale może mi się teraz ogólnie wydawać tylko.

Reply:
jw xdd